test/extended/util/configv1shim.go

```diff
@@ -0,0 +1,1022 @@
+		// only metadata.Name and metadata.Namespaces are supported for CRDs
+		// Infrastructure is cluster scoped -> no metadata.namespace
+		if watchSel.Matches(fields.Set(map[string]string{"metadata.name": item.Name})) {
+			watcher.Action(watch.Added, &obj)
+		}
+		objs = append(objs, &obj)
+	}
+
+	go func() {
+		err := watcher.Follow(resourceWatcher, objs...)
+		if err != nil {
+			klog.Errorf("Config shim fake watcher returned prematurely: %v", err)
+		}
+		watcher.Stop()
+	}()
+
+	return watcher, nil
+}
+
+func (c *ConfigV1InfrastructuresClientShim) Patch(ctx context.Context, name string, pt types.PatchType, data []byte, opts metav1.PatchOptions, subresources ...string) (*apiconfigv1.Infrastructure, error) {
+	_, err := c.fakeConfigV1InfrastructuresClient.Get(ctx, name, metav1.GetOptions{})
+	if err == nil {
+		return nil, &OperationNotPermitted{Action: "patch"}
+	}
+	if apierrors.IsNotFound(err) {
+		return c.configV1InfrastructuresClient.Patch(ctx, name, pt, data, opts, subresources...)
+	}
+	return nil, err
+}
+
+func (c *ConfigV1InfrastructuresClientShim) Apply(ctx context.Context, infrastructure *applyconfigv1.InfrastructureApplyConfiguration, opts metav1.ApplyOptions) (*apiconfigv1.Infrastructure, error) {
+	// Unable to determine existence of a static manifest
+	if infrastructure == nil || infrastructure.Name == nil {
+		return c.configV1InfrastructuresClient.Apply(ctx, infrastructure, opts)
+	}
+	_, err := c.fakeConfigV1InfrastructuresClient.Get(ctx, *infrastructure.Name, metav1.GetOptions{})
+	if err == nil {
+		return nil, &OperationNotPermitted{Action: "apply"}
+	}
+	if apierrors.IsNotFound(err) {
+		return c.configV1InfrastructuresClient.Apply(ctx, infrastructure, opts)
+	}
+	return nil, err
+}
+
+func (c *ConfigV1InfrastructuresClientShim) ApplyStatus(ctx context.Context, infrastructure *applyconfigv1.InfrastructureApplyConfiguration, opts metav1.ApplyOptions) (*apiconfigv1.Infrastructure, error) {
+	// Unable to determine existence of a static manifest
+	if infrastructure == nil || infrastructure.Name == nil {
+		return c.configV1InfrastructuresClient.ApplyStatus(ctx, infrastructure, opts)
+	}
+	_, err := c.fakeConfigV1InfrastructuresClient.Get(ctx, *infrastructure.Name, metav1.GetOptions{})
+	if err == nil {
+		return nil, &OperationNotPermitted{Action: "applystatus"}
+	}
+	if apierrors.IsNotFound(err) {
+		return c.configV1InfrastructuresClient.ApplyStatus(ctx, infrastructure, opts)
+	}
+	return nil, err
+}
+
+type ConfigV1NetworksClientShim struct {
+	fakeConfigV1NetworksClient configv1.NetworkInterface
+	configV1NetworksClient     configv1.NetworkInterface
+}
+
+var _ configv1.NetworkInterface = &ConfigV1NetworksClientShim{}
+
+func (c *ConfigV1NetworksClientShim) Create(ctx context.Context, infrastructure *apiconfigv1.Network, opts metav1.CreateOptions) (*apiconfigv1.Network, error) {
```

Comment:
nit: still wrong variable names (`infrastructure`)

Reply:
and in other methods as well

Reply:
Done